File: Models/MainTemplateVariants.cs
namespace XmlEditorUi;

public static class MainTemplateVariants
{
    public const string Standard = "Standard (Umschulung)";
    public const string Externenpruefung = "Externenprüfung";

    public const string StandardFileName = "Main.xml";
    public const string ExternFileName = "Main - Externenprüfung.xml";

    public static IReadOnlyList<string> All { get; } = [Standard, Externenpruefung];

    public static string GetFileName(string variant) =>
        IsExternenpruefung(variant) ? ExternFileName : StandardFileName;

    public static bool IsExternenpruefung(string? variant) =>
        string.Equals(variant, Externenpruefung, StringComparison.OrdinalIgnoreCase);
}

File: flutter/lib/domain/templates/main_template_variants.dart
class MainTemplateVariants {
  static const standard = 'Standard (Umschulung)';
  static const externenpruefung = 'Externenprüfung';

  static const standardFileName = 'Main.xml';
  static const externFileName = 'Main - Externenprüfung.xml';

  static const List<String> all = [standard, externenpruefung];

  static String getFileName(String variant) =>
      isExternenpruefung(variant) ? externFileName : standardFileName;

  static bool isExternenpruefung(String? variant) =>
      variant != null &&
      variant.toLowerCase() == externenpruefung.toLowerCase();
}
